### PR TITLE
Finished markdown collapsible folders ticket

### PR DESCRIPTION
**Pull request by Manpreet-Bhatti**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,51 +1,51 @@
...
+  const GITHUB_API_BLOBS_SUFFIX = '/git/blobs';
+  const GITHUB_API_TREES_SUFFIX = '/git/trees';
+  const GITHUB_API_TREES_LANGUAGES = '/languages';
+  const GITHUB_API_TREES_CONTRIBUTORS = '/contributors';
+  const WITH_RECURSIVE_PARAMETER = '?recursive=true';
+  const NPM_API_VERSION2 = 'https://api.npms.io/v2';
   const handleExampleGoButtonPress = async () => {
-    const url = "https://github.com/cheapreats/auto-readme-docs";
-    const pathArray = url.split("/");
+    const url = 'https://github.com/cheapreats/auto-readme-docs';
```
Comment (ralph-dev) [inline src/App.tsx:47]: Should this be here or is this a const?

File: src/App.tsx [hunk 2]
```diff
@@ -201,23 +197,21 @@ const App: React.FC = () => {
...
+                  <ContentSection>{line}</ContentSection>
+                </LightBGColor>
+              )))}
             </div>
           </div>
           <div className="row">
             <CenteredCol className="col">
               <CopyToClipboard
-                text={`<big><pre>\n${repoLanguages.join("\n")}\n</pre></big>`}
+                text={`<big><pre>\n${repoLanguages.join('\n')}\n</pre></big>`}
```
Comment (ralph-dev) [inline src/App.tsx:214]: Should'nt this pre and big be a const prefix ans suffix. @sahandfarahani

File: src/utils/generateMarkDownTree/generateMarkDownTree.ts [hunk 3]
```diff
@@ -1,27 +1,75 @@
...
+  treeCore: Core[],
+  treePath: string,
+): void => {
+  const splitTreePath = treePath.split('/');
+  /* folder wrapping the the file/folder that need's to have </details> tag added on to */
+  const folderWrappingFile = splitTreePath[splitTreePath.length - 2];
+
+  for (let i = 0; i < treeCore.length; i += 1) {
+    if (
+      treeCore[i].path.split('/')[treeCore[i].path.split('/').length - 1]
```
Comment (ralph-dev) [inline src/utils/generateMarkDownTree/generateMarkDownTree.ts:39]: Abstract into a variable so you don't repeat this 30 times

File: src/utils/generateMarkDownTree/generateMarkDownTree.ts [hunk 4]
```diff
@@ -1,27 +1,75 @@
...
+const addBlockquoteDetailsTag = (
+  motherTreeCore: Core[],
+  treeCore: Core[],
+  treePath: string,
+): void => {
+  const splitTreePath = treePath.split('/');
+  /* folder wrapping the the file/folder that need's to have </details> tag added on to */
+  const folderWrappingFile = splitTreePath[splitTreePath.length - 2];
+
+  for (let i = 0; i < treeCore.length; i += 1) {
```
Comment (ralph-dev) [inline src/utils/generateMarkDownTree/generateMarkDownTree.ts:37]: Must this be a for loop or can we use a Map or reduce?

File: src/utils/generateMarkDownTree/generateMarkDownTree.ts [hunk 5]
```diff
@@ -33,15 +81,13 @@ export const generateMarkDownTree = (
     deepClonedTreeCore.forEach(
       (core: { comment: string; treeCore: Core[]; path: string }) => {
         const { path } = core;
-        let comment = "";
+        let comment = '';
         if (core.comment) {
-          comment = core.comment;
+          comment = `<span>${core.comment}</span>`;
```
Comment (ralph-dev) [inline src/utils/generateMarkDownTree/generateMarkDownTree.ts:86]: Magic tag

File: src/utils/generateMarkDownTree/generateMarkDownTree.ts [hunk 6]
```diff
@@ -33,15 +81,13 @@ export const generateMarkDownTree = (
...
+        } else if (withAutoComments) {
+          comment = `<span>${getAutoGeneratedCommentForPath(path)}</span>`;
```
Comment (ralph-dev) [inline src/utils/generateMarkDownTree/generateMarkDownTree.ts:88]: Magic tag

Duplicate code.

File: src/utils/generateMarkDownTree/generateMarkDownTree.ts [hunk 7]
```diff
@@ -54,39 +100,67 @@ export const generateMarkDownTree = (
...
+            && deepClonedTreeCore
+            && deepClonedTreeCore[0] === core
+          ) {
+            curLine += '</summary><blockquote>';
+          }
+          curLine += `<details><summary>${icon}${hyperLink} ${commentAlignment}${comment}`;
+        } else if (
+          deepClonedTreeCore
+          && core.treeCore.length >= 0
+          && curDepth > 0
```
Comment (ralph-dev) [inline src/utils/generateMarkDownTree/generateMarkDownTree.ts:122]: 0 represents ROOT?

Comment (Manpreet-Bhatti) [reply src/utils/generateMarkDownTree/generateMarkDownTree.ts:122]: Yup, changed this into const to stay away from magic num

**Comment by ralph-dev** [comment]
@sahandfarahani just have ONE parameter that's an TYPESCRIPT Enum that does
OPEN, CLOSE, BOTH for your wrap function!

**Comment by sahandfarahani** [comment]
> @sahandfarahani just have ONE parameter that's an TYPESCRIPT Enum that does
> OPEN, CLOSE, BOTH for your wrap function!

@ralph-dev Done

**Comment by ralph-dev** [comment]
> > @sahandfarahani just have ONE parameter that's an TYPESCRIPT Enum that does
> 
> > OPEN, CLOSE, BOTH for your wrap function!
> 
> 
> 
> @ralph-dev Done

Use typescript Enums instead of a CONST if there is only specific Tags that can be used or if auto complete is an experience you want to provide to future developers.

**Comment by ralph-dev** [comment]
Otherwise this is good

**Comment by ralph-dev** [comment]
@sahandfarahani  Use a switch statement you define all three states btw for code clarity.

**Comment by sahandfarahani** [comment]
> @sahandfarahani Use a switch statement you define all three states btw for code clarity.

@ralph-dev Sorry I don't know what you mean here

**Pull request merged**